### PR TITLE
Publish cube revisions

### PR DESCRIPTION
**Pull request by tpluscode**
This PR should pretty much complete the publishing feature:

1. A "revision" number is appended to cube URIs.
2. When a job completes, the last published revision is incremented on project
3. Cube data is stored at a predefined named graph, stored by project. 
   - append and not replace
4. Both the output graph as well the revision are stored with job to allow consistent re-runs

### CODE REVIEW

File: ui/tests/e2e/specs/project-mapping.spec.ts [hunk 1]
```diff
@@ -18,6 +18,11 @@ describe('CSV mapping flow', () => {
       .find('input')
       .type('My project')
 
+    cy.contains('.label', 'Publish graph')
+      .next()
+      .find('input')
+      .type('http://example.com/published-cube')
+
```
Comment (tpluscode) [inline ui/tests/e2e/specs/project-mapping.spec.ts:25]: @martinmaillard what's missing here? it did not seem to find or type in the input field

Comment (martinmaillard) [reply ui/tests/e2e/specs/project-mapping.spec.ts:25]: I need to try it out

**Comment by martinmaillard** [comment]
It seems like the revision number is also applied to properties (the ones that are created out of the project namespace). I think that's wrong.

File: ui/tests/e2e/specs/project-mapping.spec.ts (hunk 1 above)
Comment (tpluscode) [reply ui/tests/e2e/specs/project-mapping.spec.ts]: Thanks @martinmaillard

**Review by lucafurrer — APPROVED** [review]

File: apis/core/bootstrap/shapes/cube-project.ts [hunk 2]
```diff
@@ -37,6 +37,16 @@ ${shape('cube-project/create')} {
...
+    ${sh.property} [
+      ${sh.name} "Publish graph" ;
+      ${sh.description} "A named graph URI where cubes will be published from this project" ;
+      ${sh.path} ${cc.publishGraph} ;
+      ${sh.minLength} 1 ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+      ${sh.nodeKind} ${sh.IRI} ;
+      ${sh.order} 40 ;
+    ] ;
```
[the remark this answers is not in the document]
Comment (tpluscode) [reply apis/core/bootstrap/shapes/cube-project.ts:49]: They don't. Namespace is the base URI for cube. `publishGraph` is where we write the cube

File: apis/core/lib/domain/job/update.ts [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
 
 interface JobUpdateParams {
   resource: GraphPointer<NamedNode>
   store?: ResourceStore
 }
 
+function isPublishJob(job: Job): job is PublishJob {
+  return job.types.has(cc.PublishJob)
+}
+
```
[the remark this answers is not in the document]
Comment (tpluscode) [reply apis/core/lib/domain/job/update.ts:19]: Or maybe not. I would not like to see a bunch of `isFooBar` on the job object. But I move it to `@cube-creator/model/Job`

**Review by martinmaillard — APPROVED** [review]

**Pull request merged**